packages/report-server/src/auth/middleware.ts

```diff
@@ -1,20 +1,56 @@
 import { UnauthenticatedError } from '@tupaia/utils';
 import { AccessPolicy } from '@tupaia/access-policy';
+import { getUserIDFromToken } from '@tupaia/auth';
 import { ReportsRequest } from '../types';
 
+const authenticateUser = async (req: ReportsRequest) => {
+  const authHeader = req.headers.authorization || req.headers.Authorization;
+  if (!authHeader) {
+    throw new UnauthenticatedError('No authorization header provided - must be Bearer Auth Header');
+  }
+
+  if (authHeader.startsWith('Bearer')) {
+    return authenticateBearerAuthHeader(authHeader);
+  } else if (authHeader.startsWith('Basic')) {
+    return authenticateBasicAuthHeader(req, authHeader);
+  }
+
+  throw new UnauthenticatedError('Could not authenticate with the provided token');
+}
+
+const authenticateBearerAuthHeader = async (authHeader: string) => {
+  // Use the user account provided in the auth header if present
+  const tokenUserID = getUserIDFromToken(authHeader);
+  if (tokenUserID) {
+    return tokenUserID;
+  }
+
+  throw new UnauthenticatedError('Could not authenticate with the provided access token');
+}
+
+const authenticateBasicAuthHeader = async (req: ReportsRequest, authHeader: string) => {
+  const usernamePassword = Buffer.from(authHeader.split(' ')[1], 'base64').toString();
+  const [username, password] = usernamePassword.split(':');
```

Comment:
I'm a little worried that this may fail if the user has a `:` in their password? I think it's safe for us to assume that `:` can't be in the username since it's invalid in an email.

Perhaps something like: 
```
if (!usernamePassword.includes(':')) {
  throw new UnauthenticatedError('Invalid Basic auth credentials');
}
const username = usernamePassword.split(':')[0];
const password = usernamePassword.substring(username.length, usernamePassword.length);
```

Just double check me on that `substring` logic cos my small brain always screws up string indexes 😝

Reply:
Oh man... nice catch 🤦

Reply:
Fixed!